mk/llvm.mk

```diff
@@ -80,10 +80,17 @@ endif
 
 
 # LLVM linkage:
+# Note: Filter with llvm-config so that optional targets which aren't present
+# don't cause errors (ie PNaCl's target is only present within PNaCl's LLVM
+# fork).
 LLVM_LINKAGE_PATH_$(1):=$$(abspath $$(RT_OUTPUT_DIR_$(1))/llvmdeps.rs)
 $$(LLVM_LINKAGE_PATH_$(1)): $(S)src/etc/mklldeps.py $$(LLVM_CONFIG_$(1))
-	$(Q)$(CFG_PYTHON) "$$<" "$$@" "$$(LLVM_COMPONENTS)" "$$(CFG_ENABLE_LLVM_STATIC_STDCPP)" \
-		$$(LLVM_CONFIG_$(1)) "$(CFG_STDCPP_NAME)"
+	$(Q)$(CFG_PYTHON) "$$<" "$$@" "$(filter $(shell \
+				$(CFG_LLVM_ROOT)/bin/llvm-config$(X_$(CFG_BUILD)) \
+					--components), \
```

Comment:
Can this be replaced with a variable defined below to avoid re-invoking llvm-config?